### ISSUE TITLE
Abbreviate journal titles using general rules

### ISSUE THREAD

**Issue by Phidica** [nschloe/betterbib#130]
Rather than maintain a lookup table of journal names and their standard abbreviation, which obviously must be ever changing, why not instead use the formal [List of Title Word Abbreviations (LTWA)](https://www.issn.org/services/online-services/access-to-the-ltwa/). Then, we look up each word of the journal name and abbreviate according to the standard for that word.

This seems to me like the smarter, more generalised approach to implementing journal name shortening.

**Comment by Phidica** [comment]
I've realised this doesn't work with `--long-journal-names` because in that case we need to expand abbreviated names back, and there is degeneracy in the abbreviations. We would still have to maintain a list of full journal titles to enable that.

I still believe the incorporation of the LTWA would be good though, as it would enable betterbib to create the standard abbreviation for journals it hasn't seen before.

**Comment by nschloe** [comment]
Not a very pressing problem, but definitely sounds interesting. PRs welcome!

**Comment by dwhswenson** [comment]
Comment for anyone who works on this: I'm not sure exactly how [ISO 4](https://en.wikipedia.org/wiki/ISO_4) phrases things, but direct word-by-word abbreviation does not give the right answers (single-word titles are probably the exceptions).

That is, you get "Nature Methods" -> *Nat. Methods*, but "Nature" -> *Nature*; and "Journal of Polymer Science Part A" -> *J. Polym. Sci. A*, but "Science" -> *Science*.

**Comment by thorade** [comment]
related: https://tex.stackexchange.com/questions/33441/how-to-abbreviate-journal-name-in-citation

**Comment by nschloe** [comment]
Apparently, there is a project which handles ISO-4 abbreviations, https://github.com/marcinwrochna/abbrevIso. Unlikely that betterbib will ever use it because, as stated above, we'll also need unabbrevation. For this, we have to keep a list around, so why not use that for abbreviation as well.

I'll close this issue, but if anyone ever comes up with a PR, I might consider it.

**Comment by Phidica** [comment]
There are use cases that do not require unabbreviation. For instance, when I opened this issue, I was using betterbib as follows:
```sh
betterbib-format main.bib.in | betterbib-journal-abbrev --extra-abbrev-file journals.json - main.bib
```
where `main.bib.in` was the bib file I managed myself, in which I always wrote full journal names, and I treated `main.bib` as an auto-generated file, which was ultimately passed to BibLaTeX and biber. I had some journals that weren't in the built-in list so I used the website I linked in my first comment to manually define those abbreviations in `journals.json`. 

It just seemed to me that there shouldn't be a need for me to go to that website and manually type in each word of a journal name, individually, to define its abbreviated form, when a script could do the same.

However, I understand if this use case is not within the intended scope of betterbib. Being that there is an alternative library which you've linked that can achieve journal name abbreviation, perhaps in my future I will write a simple script that parses BibTeX files and performs the substitutions using that library, and I won't need betterbib at all.
